mergeByPlatform.py

```diff
@@ -14,20 +15,31 @@
              ('IlluminaHiSeq_miRNASeq', 'miRNAExp', 'bcgsc.ca_PANCAN_IlluminaHiSeq_miRNASeq.miRNAExp.tsv'),
              ('HumanMethylation27','betaValue', 'jhu-usc.edu_PANCAN_HumanMethylation27.betaValue.tsv'),
              ('HumanMethylation450', 'betaValue', 'jhu-usc.edu_PANCAN_HumanMethylation450.betaValue.tsv'),
-             #These are bed files
+             #These are bed and seg files
              ('IlluminaHiSeq_DNASeqC', 'cna', 'hms.harvard.edu_PANCAN_IlluminaHiSeq_DNASeqC.cna.bed'),
-             ('Genome_Wide_SNP_6', 'cna', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna.bed'),
-             ('Genome_Wide_SNP_6', 'cna_nocnv', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_nocnv.bed'), 
-             ('Genome_Wide_SNP_6', 'cna_nocnv_probecount', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_nocnv_probecount.bed'),
-             ('Genome_Wide_SNP_6', 'cna_probecount', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_probecount.bed')]
-             #MSI,
-             #Maf
-
+             ('Genome_Wide_SNP_6', 'cna', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna.seg'),
+             ('Genome_Wide_SNP_6', 'cna_nocnv', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_nocnv.seg'), 
+             ('Genome_Wide_SNP_6', 'cna_nocnv_probecount', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_nocnv_probecount.seg'),
+             ('Genome_Wide_SNP_6', 'cna_probecount', 'broad.mit.edu_PANCAN_Genome_Wide_SNP_6.hg19.cna_probecount.seg')]
+             #MSI,Maf
+# Generate string of unique platforms from platforms array.
+availPlatforms = '\n'.join(set([(x[0]+'\t') for x in platforms]))
+# Argument parser to allow user to indicate with synapse project to merge files from, which project to load the merged file into, 
+# and an optional platform argument if the user only has files which are subset of total platforms. 
+parser = argparse.ArgumentParser()
+parser.add_argument('benefactorId',help='ID of synapse project to merge files from.')
+parser.add_argument('parentId',help='ID of synapse project to add merged file to.')
+parser.add_argument('filePath',help='Local filepath to write merged files to.',type=str)
+parser.add_argument('-p','--platforms',nargs='*',help='If merging subset of platform type, add platform(s) name after option. \
+		    Available platforms' + '\n' +  availPlatforms,type=str)
+args = parser.parse_args()
+if args.platforms is not None:
+	platforms = [x for x in platforms if x[0] in args.platforms]	
+QUERY_STR = "select * from file where benefactorId==" + ("'{0}'").format(args.benefactorId)
 
-QUERY_STR = "select * from file where benefactorId=='syn2812961'"
 
 def isUptodate(name, files):
-    q = syn.chunkedQuery("select id from file where name=='%s' and parentId=='%s'" %(name, PARENT_ID))
+    q = syn.chunkedQuery("select id from file where name=='%s' and parentId=='%s'" %(name, args.parentId))
     try:
         id =  q.next()['file.id']
     except StopIteration:
```

Comment:
Lines 42-45 can be replaced with a call to syn._findEntityIdByNameAndParent(name, parent) instead which returns None if the file doesn't exist.